### PR TITLE
[CHAT-2048] deactivate & delete User helper method

### PR DESCRIPTION
**Pull request by gumuz**
## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] Code changes are tested

## Changelog

-

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -2305,6 +2305,28 @@ export class StreamChat<
...
 
+  /** deactivateAndDeleteUser - convenience method that both deactivates and deletes a user
+   *
+   * @param {string} userID target user ID
+   * @param {object} [options] options for both deactivateUser and deleteUser
+   * @returns {Promise<APIResponse>}
+   */
+  async deactivateAndDeleteUser(
+    userID: string,
+    options?: {
```
Comment (ferhatelmas) [inline src/client.ts:2316]: It is better to extract these options to types and combine here. Otherwise easy to go outdated.

File: src/client.ts [hunk 2]
```diff
@@ -2305,6 +2305,28 @@ export class StreamChat<
...
+      delete_conversation_channels?: boolean;
+      hard_delete?: boolean;
+      mark_messages_deleted?: boolean;
+    },
+  ) {
+    await this.deactivateUser(userID, {
+      created_by_id: options?.created_by_id,
+      mark_messages_deleted: options?.mark_messages_deleted,
+    });
+    return await this.deleteUser(userID, options);
```
Comment (AnatolyRugalev) [inline src/client.ts:2327]: What happens when we do `mark_messages_deleted` twice?

Comment (ferhatelmas) [reply src/client.ts:2327]: Should be noop but good use-case test for QA.

**Review by AnatolyRugalev — APPROVED** [review]

File: src/client.ts [hunk 3]
```diff
@@ -2305,6 +2305,28 @@ export class StreamChat<
...
+  async deactivateAndDeleteUser(
+    userID: string,
+    options?: {
+      created_by_id?: string;
+      delete_conversation_channels?: boolean;
+      hard_delete?: boolean;
+      mark_messages_deleted?: boolean;
+    },
+  ) {
+    await this.deactivateUser(userID, {
```
Comment (ferhatelmas) [inline src/client.ts:2323]: Also, should we fail here if user is already deactivated? I think not.

Comment (mahboubii) [reply src/client.ts:2323]: yea it's a noop

**Pull request closed without merging**